File: backend/models/video.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Video struct {
	gorm.Model
	Filename    string    `json:"filename"`
	Path        string    `json:"path"`
	Size        int64     `json:"size"`
	Duration    int       `json:"duration"`
	Status      string    `json:"status"` // uploaded, processing, ready
	UploadedAt  time.Time `json:"uploaded_at"`
	ProcessedAt time.Time `json:"processed_at"`
} 

File: web/app/Models/Video.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Video extends Model
{
    protected $fillable = [
        'filename',
        'path',
        'size',
        'duration',
        'status',
        'uploaded_at',
        'processed_at',
    ];

    protected $casts = [
        'uploaded_at' => 'datetime',
        'processed_at' => 'datetime',
    ];
} 